moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1312,7 +1313,23 @@ void TrajectoryExecutionManager::executeThread(const ExecutionCompleteCallback&
 
   // only report that execution finished successfully when the robot actually stopped moving
   if (last_execution_status_ == moveit_controller_manager::ExecutionStatus::SUCCEEDED)
-    waitForRobotToStop(*trajectories_[i - 1]);
+  {
+    execution_trajectories_mutex_.lock();
+    if (!execution_complete_ && trajectories_.size() >= i)
```

Comment:
Can the condition `trajectories_.size() >= i` become false at all? That's only possible if clear() is called in between.
As we hold the `execution_state_mutex_` this cannot happen anymore.
```suggestion
    if (!execution_complete_)
```